src/sentry/api/serializers/models/release.py

```diff
@@ -197,6 +189,44 @@ def get_attrs(self, item_list, user, *args, **kwargs):
                 .values('release_id').annotate(new_groups=Sum('new_groups'))
                 .values_list('release_id', 'new_groups')
             )
+        return tags, group_counts_by_release
+
+    def __get_release_data_with_environment(self, project, item_list, environment):
+        if project:
```

Comment:
In my opinion, it makes more sense to build out the base queryset here and then add the additional filters in the conditional:

```python
release_project_envs = ReleaseProjectEnvironment.objects.filter(
  release__in=item_list,
  environment=environment,
).select_related('release')
if project:
    release_project_envs = release_project_envs.filter(project=project)

```

Reply:
Oh yeah... I always forget that is an option. I tend to just rewrite it. Can do.